src/python/pants/engine/internals/graph_test.py

```diff
@@ -1950,19 +1954,43 @@ async def infer_smalltalk_dependencies(request: InferSmalltalkDependencies) -> I
             AddressInput.parse(line[1:], description_of_origin="smalltalk rule"),
         )
         for line in all_lines
-        if line.startswith("!")
+        if line.startswith("!") and not line.startswith("!!")
```

Comment:
Why was this change needed? Will it mess up existing code that does the same, or is this only a pattern specific to this test?

Reply:
Just this test. The test pretends the lines of the files are addresses (akin to the `dependencies` field) but didn't account for `!!`, so here we are accounting for `!!` since we know we're introducing that into some files.

Reply:
yea, makes sense.